dev/com.ibm.ws.jaxrs.2.1.common/src/com/ibm/ws/jaxrs20/security/LibertyAuthFilter.java

```diff
@@ -75,13 +89,157 @@ private boolean authenticate(Message m) {
     }
 
     private void handleMessage(Message message) throws UnauthenticatedException{
-        HttpServletRequest req = (HttpServletRequest) message.get(AbstractHTTPDestination.HTTP_REQUEST);
-        Method method = MessageUtils.getTargetMethod(message, () -> 
-            new AccessDeniedException("Method is not available : Unauthorized"));
-        if (RoleMethodAuthUtil.parseMethodSecurity(method, req.getUserPrincipal(), s -> req.isUserInRole(s))) {
-            return;
+        SecurityContext jaxrsSecurityContext = message.get(SecurityContext.class);
+        if (jaxrsSecurityContext != null && jaxrsSecurityContext instanceof SecurityContext) {
+            Method method = getTargetMethod(message);
+            if (parseMethodSecurity(method, jaxrsSecurityContext)) {
```

Comment:
```suggestion
            if (RoleMethodAuthUtil.parseMethodSecurity(method, jaxrsSecurityContext.getUserPrincipal(), s -> jaxrsSecurityContext.isUserInRole(s))) {
```

We shouldn't need to do the role/class/method checking here - the RoleMethodAuthUtil should do that for us - we just need to pass them the user principal (from the security context if available or the servlet) and the method from them to call to check if the user is in the role.